### ISSUE TITLE
2020.0 CI release : incorrect song titles on importing plain text files

### ISSUE THREAD

**Issue by DeeTigIT** [quelea-projection/Quelea#167]
Hello,

The 2019.1 release seems to have an issue with the song title determination on importing plain text files. I added 2 examples:
- These are UTF-8 files (no BOM)
- The title is specified with keyword on the first line
These songs enter in de db with the wrong title.
Already upon the import selection screen (where one enables or disables songs) the wrong title is shown.

![S1 - 2019 1](https://user-images.githubusercontent.com/39883796/61556279-b105ae80-aa61-11e9-8c41-26b189896b6e.JPG)

![S2 - 2019 0](https://user-images.githubusercontent.com/39883796/61556291-b6fb8f80-aa61-11e9-8c53-e46bb2d5d138.JPG)

[OPWEKKING-0791 - Almachtige.txt](https://github.com/quelea-projection/Quelea/files/3412216/OPWEKKING-0791.-.Almachtige.txt)

[DIVERSEN-0081 - Great Are You Lord (All Sons & Daughters) (VT).txt](https://github.com/quelea-projection/Quelea/files/3412218/DIVERSEN-0081.-.Great.Are.You.Lord.All.Sons.Daughters.VT.txt)

Regards,

Dirk

**Comment by DeeTigIT** [comment]
Maybe there's a relation here with the 2019.1 changelog item:
 "Fix plain text parser to use file name as title if no title is given in document"?
In our set of 3444 songs, there are 65 songs with incorrect title.

**Comment by DeeTigIT** [comment]
I checked some of the songs with wrongly imported song title and it turns out that : **each of these songs has a song section with size 1.** 

Then I compared the PlainTextParser.java code of the 2019.0 and 2020.0 CI releases.
There are 2 changes in the 2020.0 CI code:

- The defaultSongTitle is now deduced from the fileName and the new song is default created with this title.

- The 2nd change is in a part that checks **section size 1** in the condition:

                                if (song.getTitle().equals(**defaultSongTitle**) && section.size() == 1) {

I guess song.getTitle().equals(defaultSongTitle) is always true. So if a song has a section with size 1: the song title keyword value, as set at the beginning of the file,  is overwritten with the section line.

I think the original condition in 2019.0 is the correct one:

                                if (song.getTitle().equals(**DEFAULT_TITLE**) && section.size() == 1) {

![S1](https://user-images.githubusercontent.com/39883796/64071005-7d2cc400-cc70-11e9-853b-350990688dd2.jpg)
